packages/gatsby-source-nacelle/src/source-nodes/products.js

```diff
@@ -0,0 +1,71 @@
+const { nacelleClient } = require('../services');
+const {
+  replaceKey,
+  cacheIsInvalid,
+  hasBeenIndexedSinceLastBuild
+} = require('../utils');
+
+module.exports = async function (gatsbyApi, pluginOptions) {
+  const { actions, createContentDigest, cache } = gatsbyApi;
+  const { createNode, touchNode } = actions;
+
+  const { nacelleSpaceId, nacelleGraphqlToken } = pluginOptions;
+
+  const client = nacelleClient({
+    nacelleSpaceId,
+    nacelleGraphqlToken
+  });
+
+  try {
```

Comment:
Same as comment on `content.js`

Reply:
UPDATE: refactored to use IoC / DI to stay DRY